### PR TITLE
feat(dev): port evdev input driver

### PR DESCRIPTION
**Pull request by niklasf**
Port evdev input driver to lvgl 9, with support for multiple input devices.

* [x] Runtime config
* [x] Test with keypad
* [x] Test with pointer device

### CODE REVIEW

File: Kconfig [hunk 1]
```diff
@@ -1295,6 +1295,30 @@ menu "LVGL configuration"
...
+		config LV_EVDEV_HORIZONTAL_MIN
+			int "Minimum horizontal coordinate to transform evdev pointer events"
+			depends on LV_EVDEV_CALIBRATE
+		config LV_EVDEV_HORIZONTAL_MAX
+			int "Maximum horizontal coordinate to transform evdev pointer events"
+			depends on LV_EVDEV_CALIBRATE
+		config LV_EVDEV_VERTICAL_MIN
+			int "Minimum vertical coordinate to transform evdev pointer events"
+			depends on LV_EVDEV_CALIBRATE
+		config LV_EVDEV_VERTICAL_MAX
```
Comment (kisvegabor) [inline Kconfig]: In v9 LVGL's multi display feature should be better supported. For this these should be runtime configs. E.g. `lv_evdev_set_calibration()`

**Review by kisvegabor — COMMENTED** [review]
Thank you! 

Just one comment.

**Comment by niklasf** [comment]
Tested with mouse and keyboard. Ready for review.

**Comment by kisvegabor** [comment]
Looks good to me! Thank you for the quick update. 

I'll re-organize the docs a little bit to prepare some space for the drivers like this.

**Comment by ceggers-arri** [comment]
It looks like this doesn't work with encoders. In see some support for ```EV_REL```/```EV_ABS``` in ```_evdev_read()```, but ```lv_evdev_t::root_x```/ ```lv_evdev_t::root_y``` are not further processed for the ```LV_INDEV_TYPE_ENCODER``` case.

**Comment by kisvegabor** [comment]
How should it work with encoders? Do you mean using the mousewheel as encoder?

**Comment by ceggers-arri** [comment]
I have a "standalone" encoder (with a separate GPIO button) as evdev device under Linux.

Here is some [documentation](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/devicetree/bindings/input/rotary-encoder.txt) and the [driver](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/input/misc/rotary_encoder.c).

This is the `evtest` log when moving the encoder:
```
Event: time 1719562861.296558, type 2 (EV_REL), code 1 (REL_Y), value 1
Event: time 1719562861.296558, -------------- SYN_REPORT ------------
Event: time 1719562864.154658, type 2 (EV_REL), code 1 (REL_Y), value -1
Event: time 1719562864.154658, -------------- SYN_REPORT ------------
```

**Comment by kisvegabor** [comment]
I see! It's really not supported now, but it seems straightforward to add. Can you give a try to implement it and send a Pull request?

**Pull request merged**